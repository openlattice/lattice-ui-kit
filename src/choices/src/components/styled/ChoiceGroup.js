import styled from 'styled-components';

const ChoiceGroup = styled.div`
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fit, minmax(120px, 1fr));
  flex-wrap: wrap;
`;

export default ChoiceGroup;
