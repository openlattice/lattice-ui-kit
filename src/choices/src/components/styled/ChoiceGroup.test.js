import React from 'react';

import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import ChoiceGroup from './ChoiceGroup';

describe('ChoiceGroup', () => {
  test('row=true should yield multiple columns', () => {
    const wrapper = mount(<ChoiceGroup row />);
    expect(wrapper).toHaveStyleRule('grid-template-columns', 'repeat(auto-fit,minmax(200px,1fr))');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('row=false should yield single column', () => {
    const wrapper = mount(<ChoiceGroup />);
    expect(wrapper).toHaveStyleRule('grid-template-columns', '1fr');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
