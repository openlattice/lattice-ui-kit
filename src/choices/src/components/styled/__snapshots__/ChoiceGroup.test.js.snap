// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoiceGroup row=false should yield single column 1`] = `
.c0 {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 1fr;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<ChoiceGroup>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "ChoiceGroup-sc-1pvteyi-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "display:grid;grid-gap:10px;grid-template-columns:",
            [Function],
            ";flex-wrap:wrap;",
          ],
        },
        "displayName": "ChoiceGroup",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "ChoiceGroup-sc-1pvteyi-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</ChoiceGroup>
`;

exports[`ChoiceGroup row=true should yield multiple columns 1`] = `
.c0 {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fit,minmax(120px,1fr));
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<ChoiceGroup
  row={true}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "ChoiceGroup-sc-1pvteyi-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "display:grid;grid-gap:10px;grid-template-columns:",
            [Function],
            ";flex-wrap:wrap;",
          ],
        },
        "displayName": "ChoiceGroup",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "ChoiceGroup-sc-1pvteyi-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    row={true}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</ChoiceGroup>
`;
